### ISSUE TITLE
Unable to execute statement: cannot start a transaction within a transaction

### ISSUE THREAD

**Issue by UnEnanoMas** [matcracker/BedcoreProtect#47]
### Issue description

- Expected result: What were you expecting to happen? Dont crash i think
- Actual result: What actually happened? Crash

### Steps to reproduce the issue
IDK

### OS and versions

* BedcoreProtect: dev 385

* PocketMine-MP: 3.19.1
* PHP: 7.4
* Server OS: Linux
* Game version: W10

### Plugins
Cant share

- If you remove all plugins, does the issue still occur?
IDK

### Crashdump, backtrace or other files
```
Warning: SQLite3Stmt::execute(): Unable to execute statement: cannot start a transaction within a transaction in phar:///home/container/plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/sqlite3/Sqlite3Thread.php on line 89
[20:41:54] [Server thread/CRITICAL]: ErrorException: "Invalid argument supplied for foreach()" (EXCEPTION) in "plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await" at line 459
[20:41:54] [Server thread/CRITICAL]: #0 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(459): pocketmine\utils\Utils::errorExceptionHandler(integer 2, string[39] Invalid argument supplied for foreach(), string[123] phar:///home/container/plugins/BedcoreProtect_dev-385.phar/src/matcracker/Bedcor, integer 459, array[1])
[20:41:54] [Server thread/CRITICAL]: #1 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(217): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->reject(object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError)
[20:41:54] [Server thread/CRITICAL]: #2 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(200): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->wakeup(object Closure)
[20:41:54] [Server thread/CRITICAL]: #3 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(422): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->wakeupFlat(object Closure)
[20:41:54] [Server thread/CRITICAL]: #4 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/AwaitChild(63): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->recheckPromiseQueue(object matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\AwaitChild)
[20:41:54] [Server thread/CRITICAL]: #5 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(461): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\AwaitChild->reject(object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError)
[20:41:54] [Server thread/CRITICAL]: #6 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(217): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->reject(object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError)
[20:41:54] [Server thread/CRITICAL]: #7 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(200): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->wakeup(object Closure)
[20:41:54] [Server thread/CRITICAL]: #8 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/Await(422): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->wakeupFlat(object Closure)
[20:41:54] [Server thread/CRITICAL]: #9 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/SOFe/AwaitGenerator/AwaitChild(63): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\Await->recheckPromiseQueue(object matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\AwaitChild)
[20:41:54] [Server thread/CRITICAL]: #10 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/base/DataConnectorImpl(240): matcracker\BedcoreProtect\libs\SOFe\AwaitGenerator\AwaitChild->reject(object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError, NULL )
[20:41:54] [Server thread/CRITICAL]: #11 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/base/DataConnectorImpl(186): matcracker\BedcoreProtect\libs\poggit\libasynql\base\DataConnectorImpl->reportError(object Closure, object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError, NULL )
[20:41:54] [Server thread/CRITICAL]: #12 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/base/SqlThreadPool(112): matcracker\BedcoreProtect\libs\poggit\libasynql\base\DataConnectorImpl->matcracker\BedcoreProtect\libs\poggit\libasynql\base\{closure}(object matcracker\BedcoreProtect\libs\poggit\libasynql\SqlError)
[20:41:54] [Server thread/CRITICAL]: #13 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/base/DataConnectorImpl(268): matcracker\BedcoreProtect\libs\poggit\libasynql\base\SqlThreadPool->readResults(array[8])
[20:41:54] [Server thread/CRITICAL]: #14 plugins/BedcoreProtect_dev-385.phar/src/matcracker/BedcoreProtect/libs/poggit/libasynql/base/SqlThreadPool(65): matcracker\BedcoreProtect\libs\poggit\libasynql\base\DataConnectorImpl->checkResults()
[20:41:54] [Server thread/CRITICAL]: #15 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(113): matcracker\BedcoreProtect\libs\poggit\libasynql\base\SqlThreadPool->matcracker\BedcoreProtect\libs\poggit\libasynql\base\{closure}()
[20:41:54] [Server thread/CRITICAL]: #16 pmsrc/vendor/pocketmine/snooze/src/SleeperHandler(75): pocketmine\snooze\SleeperHandler->processNotifications()
[20:41:54] [Server thread/CRITICAL]: #17 pmsrc/src/pocketmine/Server(2155): pocketmine\snooze\SleeperHandler->sleepUntil(double 1621370514.5678)
[20:41:54] [Server thread/CRITICAL]: #18 pmsrc/src/pocketmine/Server(1992): pocketmine\Server->tickProcessor()
[20:41:54] [Server thread/CRITICAL]: #19 pmsrc/src/pocketmine/Server(1586): pocketmine\Server->start()
[20:41:54] [Server thread/CRITICAL]: #20 pmsrc/src/pocketmine/PocketMine(291): pocketmine\Server->__construct(object BaseClassLoader, object pocketmine\utils\MainLogger, string[16] /home/container/, string[24] /home/container/plugins/)
[20:41:54] [Server thread/CRITICAL]: #21 pmsrc/src/pocketmine/PocketMine(321): pocketmine\server()
[20:41:54] [Server thread/CRITICAL]: #22 pmsrc(11): require(string[71] phar:///home/container/PocketMine-MP.phar/src/pocketmine/PocketMine.php)
```

**Comment by matcracker** [comment]
Hi, I'm not able to reproduce the issue. Maybe it has been fixed in 2e7c27f but not sure. If that happens again let me know.
Thank you.

**Comment by matcracker** [comment]
The problem was caused by using two mutex instances that were both executing query transactions. Now the instance is unique.
